src/SourceNodes/sourcenode.js

```diff
@@ -360,10 +360,14 @@ class SourceNode extends GraphNode {
         //update the current time
         this._currentTime = currentTime;
 
+        if(this._state === STATE.ended ){
+            clearTexture(this._gl, this._texture);
```

Comment:
we don't really want to be calling clearTexture for every RAF() - waste of resources?

Reply:
there is a defensive if inside `cleartexture`, it should protect us from this